### PR TITLE
feat: add TypeScript definitions

### LINKED ISSUE #79: TypeScript definitions
Do TypeScript definitions exist for Smooth UI?

### PR DESCRIPTION
**Pull request by Ermakoy**
closes #79

### CODE REVIEW

**Comment by neoziro** [comment]
Awesome! Thanks! Nothing to add in `package.json`?

@theKashey, could you give a look if types look right for you? I prefer to have a second look and I am not very good in TypeScript 🙏

**Comment by Ermakoy** [comment]
@theKashey 
I read that i can add types to package by putting `"types": "typedir/index.d.ts"` in package.json
But i don't know how to test that definition is shipped with builded package. I have lookup for other libraries with external TS typings (like postcss) and they somehow get their `d.ts` file shipped with other package files.
We can add `"types": "dist/index.d.ts"` into -em and -sc package.json and include definition file in rollup build phase

**Comment by neoziro** [comment]
@Ermakoy are you still working on it?

**Comment by Ermakoy** [comment]
@neoziro
Yep. I have solution, but have not committed it yet

**Comment by Ermakoy** [comment]
@neoziro 
Came up with this ^

File: packages/core-sc/package.json [hunk 1]
```diff
@@ -34,5 +35,8 @@
     "react": ">=16.3.0",
     "react-dom": ">=16.3.0",
     "styled-components": "^4.0.0"
+  },
+  "devDependencies": {
```
Comment (theKashey) [inline packages/core-sc/package.json]: Not needed here.

**Comment by theKashey** [comment]
> they somehow get their d.ts file shipped with other package files.

`d.ts` resolution is separated from "normal" resolution. You may place `index.d.ts` in the `rootDir`, you may place `server.d.ts` to handle `mypackage/server`, even if "the real file" would be `mypackage/server/index.js`. And you always could use `types` field in the package.json.

_DefinitelyTyped_ recommends storing your types in `/types` dir, but... that's quite strange rule. If your project is written in TS - your types would be stored next to your js files, thus `dist` is a good location.

File: packages/core-sc/package.json (hunk 1 above)
Comment (Ermakoy) [reply packages/core-sc/package.json]: Oh didn't notice it

**Comment by Ermakoy** [comment]
@neoziro @theKashey 
Any updates?

**Comment by neoziro** [comment]
Thanks for the review and thanks for you work! I will merge it and publish it!

**Pull request merged**